src/dmd/expressionsem.d

```diff
@@ -11542,7 +11546,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
 
         // lower some array comparisons to object.__equals(e1, e2)
-        if (needsArrayLowering || (t1.ty == Tarray && t2.ty == Tarray))
+        if (needsArrayLowering || (t1.ty == Tarray && t2.ty == Tarray) || (t1.ty == Tsarray && t2.ty == Tsarray))
```

Comment:
This is most likely too much, lowering *all* equality comparisons of static array pairs.

Reply:
That's because this test otherwise doesn't work:

```D
struct VertexB
{
    float[2] pos;
}

struct Hold
{
    VertexB v;
}

extern(C) int main()
{
    Hold[1] h_a; 
    Hold[1] h_b;
    assert(h_a != h_b);
}
```